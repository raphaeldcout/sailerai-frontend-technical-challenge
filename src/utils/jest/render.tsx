import { FC, ReactElement, ReactNode } from 'react';

import { ThemeProviderSailer } from '@/styles/withThemeProvider';
import { render, RenderOptions } from '@testing-library/react';

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => (
  <ThemeProviderSailer>{children}</ThemeProviderSailer>
);

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };

