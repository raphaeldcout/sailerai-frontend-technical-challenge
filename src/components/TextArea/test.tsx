import '@testing-library/jest-dom';

import { render } from '@/utils/jest/render';

import { TextArea } from '.';

describe('TextArea', () => {
  test('should match snapshot', () => {
    const { container } = render(<TextArea value='' handleInputChange={jest.fn()} />);

    expect(container).toMatchSnapshot();
  });
});
