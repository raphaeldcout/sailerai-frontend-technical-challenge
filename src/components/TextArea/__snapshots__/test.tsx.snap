// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea should match snapshot 1`] = `
<div>
  <div
    class="sc-4ab968c3-0 jkKxcG"
  >
    <textarea
      class="sc-4ab968c3-1 kVwsHd"
      placeholder="Digite sua mensagem..."
      rows="1"
    />
  </div>
</div>
`;
